src/Components/Users/UserProfile.tsx

```diff
@@ -0,0 +1,421 @@
+import { useDispatch, useSelector } from "react-redux";
+import { getUserDetails, updateUserDetails } from "../../Redux/actions";
+import { PhoneNumberField, SelectField, TextInputField } from "../Common/HelperInputFields";
+import { parsePhoneNumberFromString } from 'libphonenumber-js';
+import { validateEmailAddress } from "../../Common/validation";
+import * as Notification from "../../Utils/Notifications.js";
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            width: '100%',
+        },
+        heading: {
+            fontSize: theme.typography.pxToRem(15),
+            flexBasis: '33.33%',
+            flexShrink: 0,
+        },
+        secondaryHeading: {
+            fontSize: theme.typography.pxToRem(15),
+            color: theme.palette.text.secondary,
+        },
+    }),
+);
+const initForm: any = {
+    first_name: "",
+    last_name: "",
+    age: "",
+    gender: "",
+    email: "",
+    phone_number: "",
+};
+
+const initError = Object.assign({}, ...Object.keys(initForm).map(k => ({ [k]: "" })));
+
+const initialState = {
+    form: { ...initForm },
+    errors: { ...initError },
+};
+
+const editFormReducer = (states = initialState, action: any) => {
+    switch (action.type) {
+        case "set_form": {
+            return {
+                ...states,
+                form: action.form
+            };
+        }
+        case "set_error": {
+            return {
+                ...states,
+                errors: action.errors
+            };
+        }
+        default:
+            return states;
+    }
+};
+export default function UserProfile() {
+    const [states, dispatch] = useReducer(editFormReducer, initialState);
+    const classes = useStyles();
+    const state: any = useSelector(state => state);
+    const { currentUser } = state;
+    const username = currentUser.data.username;
+
+
+    const [expanded, setExpanded] = React.useState<string | false>(false);
+
+    const [isLoading, setIsLoading] = useState(false);
+    const dispatchAction: any = useDispatch();
+
+    const initialDetails: any = [{}];
+    const [details, setDetails] = useState(initialDetails);
+
+    const genderTypes = [
+        {
+            id: 0,
+            text: "Select"
+        },
+        ...GENDER_TYPES
+    ];
+
+
+    const fetchData = useCallback(
+        async (status: statusType) => {
+            setIsLoading(true);
+            const res = await dispatchAction(getUserDetails(username));
+            if (!status.aborted) {
+                if (res && res.data) {
+                    setDetails(res.data);
+                    const formData: any = {
+                        first_name: res.data.first_name,
+                        last_name: res.data.last_name,
+                        age: res.data.age,
+                        gender: res.data.gender,
+                        email: res.data.email,
+                        phone_number: res.data.phone_number,
```

Comment:
Codacy found an issue: [Identifier 'phone_number' is not in camel case.](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5811785)